controllers/platformnetwork_controller.go

```diff
@@ -402,12 +419,66 @@ func (r *PlatformNetworkReconciler) FindExistingAddressPool(client *gophercloud.
 	return pool, err
 }
 
+// Some of the network types such as mgmt use fixed names for address pools.
+// It's better to use FindAddressPoolByNetworkType for all network types other than network type "other".
+func (r *PlatformNetworkReconciler) FindAddressPoolByNetworkType(client *gophercloud.ServiceClient, instance *starlingxv1.PlatformNetwork) (pool *addresspools.AddressPool, err error) {
```

Comment:
Can we combine it with "FindExistingAddressPool"? You can differentiate the return in this method based on the network type though, the two functions have too many lines of duplication

Reply:
Simplified it by using `GetAddrPoolNameByNetworkType`, this will return "management" if network type is mgmt, otherwise it would return the instance name.
This is based on the assumption that all other network pools use same name as network name.
If the assumption is incorrect and user is allowed to use different names for networks and corresponding pools (Note: system allows us to do so) then we have to switch back to using network type to determine the corresponding network pool (as in FindAddressPoolByNetworkType), which I'll be implementing in `FindExistingAddressPool` itself.
Please let me know.